sdk/iOS/test/MSCoreDataStoreTests.m

```diff
@@ -47,6 +50,54 @@ -(void)testInit
     XCTAssertNotNil(self.store, @"store creation failed");
 }
 
+// Tests MSCoreDataStore Read, Update and Delete operations are case insensitive for the Id column.
+-(void)testReadUpdateDelete_CaseInsensitiveIdOperations
+{
+    NSError *error;
+
+    // Seed data to be inserted in the table
+    NSString * const originalItemId = @"itemid";
+    NSString * const originalText = @"original text";
+    NSArray * const originalItems = @[ @{ @"id":originalItemId, @"text":originalText } ];
+
+    // Data to be used for updating the table. The Id differs from the Id of the seed data only in case
+    NSString * const updatedItemId = @"ITEMID";
+    NSString * const updatedText = @"updated text";
+    NSArray * const updatedItems = @[ @{ @"id":updatedItemId, @"text":updatedText } ];
+
+    // Populate the table
+    [self.store upsertItems:originalItems table:TableName orError:&error];
+    XCTAssertNil(error, @"upsert failed: %@", error.description);
+
+    // Read the table using ID with different case
+    NSDictionary *item = [self.store readTable:TableName withItemId:updatedItemId orError:&error];
+    XCTAssertNil(error, @"readTable:withItemId: failed: %@", error.description);
+    XCTAssertNotNil(item, @"item should have been found in the table");
+
+    // Update the item using ID with different case
+    [self.store upsertItems:updatedItems table:TableName orError:&error];
+    XCTAssertNil(error, @"upsert failed: %@", error.description);
+
+    // Verify the original row got updated
+    item = [self.store readTable:TableName withItemId:originalItemId orError:&error];
```

Comment:
since readWithItemId returns the firstObject, I don't think this guarantees you updated the original record (it relies upon the internal implementation of local store at least).  We probably need to run readWithQuery, and verify only 1 result for that instead.  (And that result has the new data)